automation/src/main/java/org/wso2/testgrid/automation/Test.java

```diff
@@ -90,17 +95,67 @@ public TestScenario getTestScenario() {
         return testScenario;
     }
 
+    /**
+     * Returns the Pre scenario script.
+     *
+     * @return location of the script as a String.
+     */
+    public String getPreScenarioScript() {
+        return preScenarioScript;
+    }
+
+    /**
+     * Sets the Pre scenario script.
+     *
+     * @param preScenarioScript location of the script as a String.
+     */
+    public void setPreScenarioScript(String preScenarioScript) {
+        this.preScenarioScript = preScenarioScript;
+    }
+
+    /**
+     * Returns the Post scenario script.
+     *
+     * @return location of the script as a String.
+     */
+    public String getPostScenarioScript() {
+        return postScenarioScript;
+    }
+
+    /**
+     * Sets the Post scenario script.
+     *
+     * @param postScenarioScript location of the script as a String.
+     */
+    public void setPostScenarioScript(String postScenarioScript) {
+        this.postScenarioScript = postScenarioScript;
+    }
+
     /**
      * Executes the test for the given test location and deployment.
      *
      * @param testLocation location of the tests
      * @param deployment   deployment to execute the tests on
      * @throws TestAutomationException thrown when error on executing tests
      */
-    public void execute(String testLocation, Deployment deployment) throws TestAutomationException {
+    public TestScenario execute(String testLocation, Deployment deployment)
+            throws TestAutomationException, CommandExecutionException {
+        if (!StringUtil.isStringNullOrEmpty(preScenarioScript)) {
+            String result = testExecutor.executeEnvironmentScript(Paths.get(preScenarioScript), deployment);
+            if (result.contains("OK")) {
```

Comment:
Shall we return a boolean instead of a string validation?

Reply:
These changes will be done while doing this task #193